### PR TITLE
Astro deployment token docs

### LINKED ISSUE #3580: Add `api-deployment-token` docs
### Issue Type

Missing content for an existing feature

### Links to Affected Docs

https://docs.astronomer.io/astro/category/astro-deployment

### The Issue or Context

With CLI v1.25 release, a new CRUD command was added for `astro deployment token`.

- https://astronomer.productboard.com/feature-board/6963770-astro-all-features/features/23012591/detail

### Fix or Proposed Change

Add new CLI functionality to docs.

### Additional Notes

- https://github.com/astronomer/docs/pull/3579

### Required Reviewers

_No response_

### PR DESCRIPTION
**Pull request by lzdanski**
Resolves #3580 

To-do:

- [x] Add `astro deployment token list` output examples
- Add cross linking to relevant pages
  - [x] https://docs.astronomer.io/astro/deployment-api-tokens
  - [x] https://docs.astronomer.io/astro/automation-authentication

### CODE REVIEW

File: astro/cli/astro-deployment-token-create.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+| `-d`,`--description`           |    Description of the token. If the description contains a space, specify the entire description within quotes ""                                                | A string |
+| `-e`,`--expiration`           |    Expiration of the token in days. If the flag isn't used the token won't have an expiration.                                                | Must be between 1 and 3650 days. |
+| `-n`,`--name`           |    The token's name. If the name contains a space, specify the entire name within quotes `""`.                                                | String |
+| `-r`,`--role`           |    The role for the token. Possible values are `DEPLOYMENT_ADMIN` or a custom role name.                                                | Any valid Deployment role |
+
+## Examples
+
+```bash
+# create a deployment token
+astro deployment token create
```
Comment (aliotta) [inline astro/cli/astro-deployment-token-create.md:40]: add --deployment-id flag here

File: astro/cli/astro-deployment-token-list.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+:::
+
+List your Astro [Deployment API Token](deployment-api-tokens.md)
+
+To use your API token in an automated process, see [Authenticate an automation tool](automation-authentication.md).
+
+## Usage
+
+```sh
+astro deployment token list
```
Comment (aliotta) [inline astro/cli/astro-deployment-token-list.md:23]: add --deployment-id flag here

File: astro/cli/astro-deployment-token-rotate.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+:::
+
+Rotate an Astro [Deployment API Token](deployment-api-tokens.md)
+
+To use your API token in an automated process, see [Authenticate an automation tool](automation-authentication.md).
+
+## Usage
+
+```sh
+astro deployment token roate <token-ID> [flags]
```
Comment (aliotta) [inline astro/cli/astro-deployment-token-rotate.md:23]: add --deployment-id flag here

File: astro/cli/astro-deployment-token-update.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+| `-n`, `--name` | Update the Deployment's name. If the new name contains a space, specify the entire name within quotes "". | String |
+| `--resource-quota-cpu` | The Resource Quota CPU to use for the Deployment. | Numeric. The Resource Quota CPU to use for the Deployment. Example value: `10`. |
+| `--scheduler-size` | The size of Scheduler for the Deployment. | `small`, `medium`, `large` |
+| `-w`, `--workspace-id` | Workspace where the Deployment is located. | A valid Workspace ID. |
+
+## Examples
+
+```bash
+# create a deployment token
+astro deployment token update
```
Comment (aliotta) [inline astro/cli/astro-deployment-token-update.md:49]: add --deployment-id flag here

File: astro/cli/astro-deployment-token-list.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+| `--verbosity`     | The Log level                                         | `debug`, `info`, `warn`, `error`, `fatal`, or `panic`. Default is `warn`. |
+| `--workspace-id`  | The Workspace ID assigned to a Deployment.            | A valid Workspace ID                                                      |
+
+## Output
+
+| Output        | Description                                                          | Data Type |
+| ------------- | -------------------------------------------------------------------- | --------- |
+| `ID`          | The token ID.                                                        | String    |
+| `NAME`        | The name of the token.                                               | String    |
+| `DESCRIPTION` | (Optional) A description of the API Token.                           | String    |
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-list.md:40]:
```suggestion
| `DESCRIPTION` | The description of the API Token.                           | String    |
```
Column will always exist I assume

File: astro/cli/astro-deployment-token-list.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+| `SCOPE`       | Whether the API Token is for Deployment, Workspace, or Organization. | String    |
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-list.md:41]:
```suggestion
| `SCOPE`       | Whether the API Token is scoped to a Deployment, Workspace, or Organization. | String    |
```

File: astro/cli/astro-deployment-token-list.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+| `CREATED`     | How long ago the token was created in days.                          | String    |
+| `CEATED BY`   | The name of the person who created the token.                        | String    |
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-list.md:43]:
```suggestion
| `CEATED BY`   | The name of the user entity who created the token.                        | String    |
```
Could be an automated process

File: astro/cli/astro-deployment-token-rotate.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+| Option                         | Description                                                                            | Possible Values                                                                |
+| ------------------------------ | -------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
+| `-c`,`--clean-output`           |    Print only the token as output. For use of the command in scripts.                                              | N/A |
+| `-f`, `--force` | Rotate the Deployment API token without showing a warning. | N/A                                          |
+| `-t`,`--name`          | The name of the token to be rotated. If the name contains a space, specify the entire name within quotes "".          | String                                                         |
+
+## Examples
+
+```bash
+astro deployment token rotate --deployment-id=clukapi6r000008l58530cg8i [flags]
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-rotate.md:37]:
```suggestion
astro deployment token rotate --deployment-id=clukapi6r000008l58530cg8i --name "My token"
```

File: astro/cli/astro-deployment-token-rotate.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+:::
+
+Rotate an Astro [Deployment API Token](deployment-api-tokens.md)
+
+To use your API token in an automated process, see [Authenticate an automation tool](automation-authentication.md).
+
+## Usage
+
+```sh
+astro deployment token roate --deployment-id=<deployment-id> <token-ID> [flags]
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-rotate.md:23]: This usage is inconsistent with the example below. How is the token ID specified, and is it required?

File: astro/cli/astro-deployment-token-rotate.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+## Options
+
+| Option                         | Description                                                                            | Possible Values                                                                |
+| ------------------------------ | -------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
+| `-c`,`--clean-output`           |    Print only the token as output. For use of the command in scripts.                                              | N/A |
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-rotate.md:30]:
```suggestion
| `-c`,`--clean-output`           |    Print only the token as output. Use when writing scripts that run the command.                                           | N/A |
```

File: astro/cli/astro-deployment-token-update.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+| `-n`, `--name` | Update the Deployment's name. If the new name contains a space, specify the entire name within quotes "". | String |
+| `--resource-quota-cpu` | The Resource Quota CPU to use for the Deployment. | Numeric. The Resource Quota CPU to use for the Deployment. Example value: `10`. |
+| `--scheduler-size` | The size of Scheduler for the Deployment. | `small`, `medium`, `large` |
+| `-w`, `--workspace-id` | Workspace where the Deployment is located. | A valid Workspace ID. |
+
+## Examples
+
+```bash
+# create a deployment token
+astro deployment token update --deployment-id=clukapi6r000008l58530cg8i
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-update.md:49]: For whatever the eventual flags are, let's add a realistic example that configures a few different details about the token

File: astro/cli/release-notes.md [hunk 12]
```diff
@@ -16,7 +16,7 @@ import HybridBadge from '@site/src/components/HybridBadge';
 
 This document provides a summary of all changes made to the [Astro CLI](cli/overview.md). For general product release notes, go to [Astro Release Notes](release-notes.md). If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/open-support-request).
 
-- **Stable versions**: {{CLI_VER_LATEST}}, {{CLI_VER_2}}, and {{CLI_VER_3}}. See [Astro CLI release and lifecycle policy](cli/release-lifecycle-policy.md).
+- **Stable versions**: {{CLI_VER_LATEST}}, {{CLI_VER_2}}, and {{CLI_VER_3}}. See [Astro CLI release and lifecycle policy](cli/release-lifecycle-policy.md) for more information about stable CLI versions and end of life support for CLI versions.
```
Comment (jwitz) [inline astro/cli/release-notes.md:19]:
```suggestion
- **Stable versions**: {{CLI_VER_LATEST}}, {{CLI_VER_2}}, and {{CLI_VER_3}}. See [Astro CLI release and lifecycle policy](cli/release-lifecycle-policy.md) for more information about support for CLI versions.
```

**Review by jwitz — CHANGES_REQUESTED** [review]
It looks like one of the commands doesn't have the right options documented, but otherwise this is close!

File: astro/cli/astro-deployment-token-create.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+:::
+
+Create an Astro [Deployment API Token](deployment-api-tokens.md)
+
+To use your API token in an automated process, see [Authenticate an automation tool](automation-authentication.md).
+
+## Usage
+
+```sh
+astro deployment token create --deployment-id=<deployment-id>
```
[the remark this answers is not in the document]
Comment (lzdanski) [reply astro/cli/astro-deployment-token-create.md:23]: Yes, `deployment-id` is required for all commands!

File: astro/cli/astro-deployment-token-create.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+## Options
+
+| Option                         | Description                                                                            | Possible Values                                                                |
+| ------------------------------ | -------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
+| `-c`,`--clean-output`           |    Print only the token as output. For use of the command in scripts.                                                | `True` or `False` |
+| `-d`,`--description`           |    Description of the token. If the description contains a space, specify the entire description within quotes ""                                                | A string |
+| `-e`,`--expiration`           |    Expiration of the token in days. If the flag isn't used the token won't have an expiration.                                                | Must be between 1 and 3650 days. |
```
[the remark this answers is not in the document]
Comment (lzdanski) [reply astro/cli/astro-deployment-token-create.md:32]: @aliotta - there's a typo in the CLI help string if this is the case. 

```
  -e, --expiration int       Expiration of the token in days. If the flag isn't used the token won't have an expiration. Must be between 1 and 3650 days.
```

File: astro/cli/astro-deployment-token-rotate.md (hunk 9 above)
Comment (lzdanski) [reply astro/cli/astro-deployment-token-rotate.md:20]: @aliotta can you share an example? the help text isn't quite clear about the syntax

File: astro/cli/astro-deployment-token-create.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (aliotta) [reply astro/cli/astro-deployment-token-create.md:20]: yep can confirm it is required

File: astro/cli/astro-deployment-token-create.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (aliotta) [reply astro/cli/astro-deployment-token-create.md:29]: ah yeah good eyes will get a PR up for that

File: astro/cli/astro-deployment-token-rotate.md (hunk 9 above)
Comment (aliotta) [reply astro/cli/astro-deployment-token-rotate.md:20]: roate -> rotate

token-ID is not required and should go before the --deployment-id flag

Comment (aliotta) [reply astro/cli/astro-deployment-token-rotate.md:20]: example 

astro deployment token rotate MYTOKENID --deployment-id=DEPID

or

astro deployment token rotate --deployment-id=DEPID

File: astro/cli/astro-deployment-token-rotate.md [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+:::
+
+Rotate an Astro [Deployment API Token](deployment-api-tokens.md)
+
+To use your API token in an automated process, see [Authenticate an automation tool](automation-authentication.md).
+
+## Usage
+
+```sh
+astro deployment token roate --deployment-id=<deployment-id> <token-ID> [flags]
```
Comment (lzdanski) [inline astro/cli/astro-deployment-token-rotate.md:20]:
```suggestion
astro deployment token rotate <token-id> --deployment-id=<deployment-id>  [flags]
```

File: astro/cli/astro-deployment-token-update.md [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+
+```sh
+astro deployment update --deployment-id=<deployment-id> [flags]
+```
+
+## Options
+
+| Option                         | Description                                                                            | Possible Values                                                                |
+| ------------------------------ | -------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
+|`--cicd-enforcement`           |    When enabled CI/CD Enforcement where deploys to deployment must use an API Key or Token. This essentially forces Deploys to happen through CI/CD.                                                | `disable` or `enable` |
```
Comment (aliotta) [inline astro/cli/astro-deployment-token-update.md:27]: all these option look incorrect
Screenshot 2024-04-11 at 10 06 13 AM

File: astro/cli/astro-deployment-token-update.md [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+:::
+
+Update an Astro [Deployment API Token](deployment-api-tokens.md)
+
+To use your API token in an automated process, see [Authenticate an automation tool](automation-authentication.md).
+
+## Usage
+
+```sh
+astro deployment update --deployment-id=<deployment-id> [flags]
```
Comment (aliotta) [inline astro/cli/astro-deployment-token-update.md:20]: this is incorrect

Comment (aliotta) [reply astro/cli/astro-deployment-token-update.md:20]: astro deployment token update --deployment-id=<deployment-id>

Comment (lzdanski) [inline astro/cli/astro-deployment-token-update.md:20]:
```suggestion
astro deployment token update --deployment-id=MOCK_DEP_ID
```

File: astro/cli/astro-deployment-token-rotate.md [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+id: astro-deployment-token-rotate
+description: Rotate your Deployment API tokens.
+hide_table_of_contents: true
+sidebar_custom_props: { icon: 'img/term-icon.png' }
+---
+
+:::cliastroonly
+:::
+
+Rotate an Astro [Deployment API Token](deployment-api-tokens.md)
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-rotate.md:13]:
```suggestion
Rotate an Astro [Deployment API Token](deployment-api-tokens.md).
```

File: astro/cli/astro-deployment-token-update.md [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+id: astro-deployment-token-update
+description: Update your Deployment API tokens.
+hide_table_of_contents: true
+sidebar_custom_props: { icon: 'img/term-icon.png' }
+---
+
+:::cliastroonly
+:::
+
+Update an Astro [Deployment API Token](deployment-api-tokens.md)
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-update.md:13]:
```suggestion
Update an Astro [Deployment API Token](deployment-api-tokens.md).
```

File: astro/cli/astro-deployment-token-delete.md [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+
+| Option                         | Description                                                                            | Possible Values                                                                |
+| ------------------------------ | -------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
+| `-f`,`--force`           |    Delete or remove the API token without showing a warning                                                | N/A |
+| `-t`,`--name`           |    The name of the token to be deleted. If the name contains a space, specify the entire name within quotes ""                                                | String |
+
+## Examples
+
+```bash
+# delete a deployment token
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-delete.md:33]:
```suggestion
# Delete a Deployment API token
```

File: astro/cli/astro-deployment-token-list.md [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+id: astro-deployment-token-list
+description: List your Deployment API tokens.
+hide_table_of_contents: true
+sidebar_custom_props: { icon: "img/term-icon.png" }
+---
+
+:::cliastroonly
+:::
+
+List your Astro [Deployment API Token](deployment-api-tokens.md)
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-list.md:13]:
```suggestion
List all [API tokens](deployment-api-tokens.md) belonging to a Deployment.
```

File: astro/cli/astro-deployment-token-list.md [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+
+To use your API token in an automated process, see [Authenticate an automation tool](automation-authentication.md).
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-list.md:15]:
```suggestion
To use an API token in an automated process, see [Authenticate an automation tool](automation-authentication.md).
```

File: astro/cli/astro-deployment-token-delete.md [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+id: astro-deployment-token-delete
+description: Delete a Deployment API token.
+hide_table_of_contents: true
+sidebar_custom_props: { icon: 'img/term-icon.png' }
+---
+
+:::cliastroonly
+:::
+
+Delete a [Deployment API Token](deployment-api-tokens.md).
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-delete.md:13]: @aliotta Can you also use this command to remove Organization API tokens that have been assigned to a Deployment?

File: astro/cli/astro-deployment-token-update.md [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+|`-d`, `--description`           |    Updated description of the token. If the description contains a space, specify the entire description in quotes `""`.                                              | String |
+| `-t`,`--name`           |    The current name of the token. If the name contains a space, specify the entire name within quotes `""`.                                                | String |
+| `-n`, `--new-name` | The token's new name. If the new name contains a space, specify the entire name within quotes "". | String |
+| `-r`, `--role` | The role for the token. Possible values are `DEPLOYMENT_ADMIN` or a custom role name. | Any valid Deployment role. |
+
+## Examples
+
+```bash
+# The CLI prompts you to input a role for a token with Token ID assigned to a specific Deployment
+astro deployment token update TOKEN_ID --deployment-id=clukapi6r000008l58530cg8i
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-update.md:36]:
```suggestion
astro deployment token update <token-id> --deployment-id=clukapi6r000008l58530cg8i
```

File: astro/cli/astro-deployment-token-update.md [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+
+# The CLI prompts you to input a role for a token identified by its name
+astro deployment token update --deployment-id=clukapi6r000008l58530cg8i -name=TOKEN_NAME
```
Comment (jwitz) [inline astro/cli/astro-deployment-token-update.md:39]:
```suggestion
astro deployment token update --deployment-id=clukapi6r000008l58530cg8i --name="Token name"
```

**Review by jwitz — APPROVED** [review]
Core reference LGTM, but I need to follow up on our language surrounding Deployment API tokens. I realized that we don't have any documentation about assigning Org/ Workspace API tokens to a Deployment, which would change some of the language here but also requires some core doc changes. Writing an issue for that now

File: astro/cli/astro-deployment-token-delete.md (hunk 23 above)
Comment (aliotta) [reply astro/cli/astro-deployment-token-delete.md:13]: no. I have a seperate PR that adds this functionality but it adds it as a separate command.

**Review by aliotta — APPROVED** [review]

**Pull request merged**